main.go

```diff
@@ -215,8 +234,16 @@ func main() {
 			testModel.EnvironmentMatrix.IosDeviceList.IosDevices = append(testModel.EnvironmentMatrix.IosDeviceList.IosDevices, &newDevice)
 		}
 
+		timeout := configs.TestTimeout
```

Comment:
Please move the validation to a new method, to increase readability and testability.